PTMCMCSampler/PTMCMCSampler.py

```diff
@@ -431,7 +452,7 @@ def sample(
         Neff = 0
         while runComplete is False:
             iter += 1
-
+            self.comm.barrier()  # make sure all processes are at the same iteration
```

Comment:
I think this is unnecessary communication.  Using `bcast` below should guarantee that all MPI processes are always running the same step after the first loop.